src/workflows/server/server.py

```diff
@@ -748,16 +760,129 @@ async def _get_handlers(self, request: Request) -> JSONResponse:
         """
         ---
         summary: Get handlers
-        description: Returns all workflow handlers.
+        description: Returns all workflow handlers with optional filtering.
+        parameters:
+          - in: query
+            name: status
+            schema:
+              type: string
+              enum: [running, completed, failed, cancelled]
+            description: Filter by handler status
+          - in: query
```

Comment:
might make sense to make this consistent with the agent data query DSL https://developers.llamaindex.ai/python/cloud/llamaagents/agent-data-overview/#filter-dsl

I don't love that DSL, but there's not a lot of great options that don't explode in complexity. This DSL looks actually pretty similar? I would like to eventually move that agent data into the open source llamactl (or here?), so developing it out consistently should make it more re-usable and facilitate easier integration.

One difference here is that one is just using a POST to not have to get query parsing involved.

Reply:
we're already parsing json in the query. Could just continue that pattern with the other dsl?